### DISCUSSION TITLE
Search by distance threshold

### DISCUSSION THREAD

**Discussion by DrBamba** [milvus-io/milvus#5044] [Q&A and General discussion]
Is there any way to search for vectors and define the results to be all the vectors that their distances are larger then some threshold?
Instead of searching for the topK closest vectors.

I searched all over the documentation and couldn't find anything like that.

**Comment by shengjun1985** [comment]
Milvus 1.x has not supported such feature.
But can you provide some information for us to develop it?
(1) How much data do you have (row_count * dim)?
(2) What kind of distance do you usually use (L2, IP or others)?
(3) How large is your batch for a single query (nq)?
(4) What is the possible size of the result set, and can we use the current `topk` interface?

**Reply by xinjiguaike** [reply]
the same demand

**Reply by DrBamba** [reply]
1) scale of a few million vectors, dimensions between 512 and 2048
2) IP 
3) for this kind of search it'll probably be less then 100
4) not sure about this one, don't believe we can use the current topk because you don't know how many vectors fit the requested IP

**Reply by gaohang** [reply]
based on faiss range_search, milvus can do that

**Reply by DrBamba** [reply]
@gaohang but how can you access this feature through milvus?

**Reply by yhmo** [reply]
@DrBamba @gaohang 
"range search" is in plan, the design doc has been created: https://wiki.lfaidata.foundation/display/MIL/MEP+33+--+Support+Range+Search

If everything is ok, this feature will be available in next major version 2.2.
